tests/e2e/transaction/transaction_simulation.test.ts

```diff
@@ -0,0 +1,413 @@
+      // eslint-disable-next-line max-len
+      `a11ceb0b060000000801000602060a031022043208053a4e0788015208da01400c9a024c000001010102020404010001010508000000000100000302010002060506010002070701010002080901010002090a060100020403040404050403060c03050007060c060c0303050503010b0001080101080102060c03010b0001090002050b00010900030b000108010b000108010b0001080102070b000109000b0001090002070b0001090003087472616e736665720a6170746f735f636f696e04636f696e0a74776f5f62795f74776f04436f696e094170746f73436f696e087769746864726177076465706f736974056d657267650765787472616374${senderAccount.accountAddress.toStringWithoutPrefix()}00000000000000000000000000000000000000000000000000000000000000010001040003080b000b0138000c030b020b0338010201010400081a0b000a0238000c070b010a0338000c080d070b0838020d070b020b03160b061738030c090b040b0738010b050b0938010200`,
+    );
+  }, longTestTimeout);
+  describe("ED25519", () => {
+    describe("single signer", () => {
+      test("with script payload", async () => {
+        const rawTxn = await aptos.generateTransaction({
+          sender: senderAccount.accountAddress.toString(),
+          data: {
+            bytecode: signleSignerScriptBytecode,
+            type_arguments: [],
+            arguments: [new U64(1), recieverAccounts[0].accountAddress],
+          },
+        });
+        const [response] = await aptos.simulateTransaction({
+          signerPublicKey: senderAccount.publicKey,
+          transaction: rawTxn,
+        });
+        expect(response.success).toBeTruthy();
+      });
+      test("with entry function payload", async () => {
+        const rawTxn = await aptos.generateTransaction({
+          sender: senderAccount.accountAddress.toString(),
+          data: {
+            function: `0x${senderAccount.accountAddress.toStringWithoutPrefix()}::transfer::transfer`,
+            type_arguments: [],
+            arguments: [new U64(1), recieverAccounts[0].accountAddress],
+          },
+        });
+        const [response] = await aptos.simulateTransaction({
+          signerPublicKey: senderAccount.publicKey,
+          transaction: rawTxn,
+        });
+        expect(response.success).toBeTruthy();
+      });
+      test("with multisig payload", async () => {
+        const rawTxn = await aptos.generateTransaction({
+          sender: senderAccount.accountAddress.toString(),
+          data: {
+            multisigAddress: secondarySignerAccount.accountAddress,
+            function: `0x${senderAccount.accountAddress.toStringWithoutPrefix()}::transfer::transfer`,
```

Comment:
We should use `toString()` instead of `0x${toStringWithoutPrefix()}`

Reply:
it is a typing thing, `function` expects `0x${string}::${string}::${string}`, and using `toString()` typescript complains type mismatch

Reply:
Oh, that's super annoying... we should probably fix that.